src/shogun2-core/src/main/java/de/terrestris/shogun2/model/tree/TreeNode.java

```diff
@@ -0,0 +1,381 @@
+
+	/**
+	 * True if the node is expanded.
+	 */
+	private boolean expanded;
+
+	/**
+	 * Path to an image to use as an icon.
+	 */
+	private String icon;
+
+	/**
+	 * One or more space separated CSS classes to be applied to the icon
+	 * element. The CSS rule(s) applied should specify a background image to be
+	 * used as the icon.
+	 */
+	private String iconCls;
+
+	/**
+	 * Tooltip text to show on this node.
+	 */
+	private String qTip;
+
+	/**
+	 * Tooltip title.
+	 */
+	private String qTitle;
+
+
+	/**
+	 * Explicitly adding the default constructor as this is important, e.g. for
+	 * Hibernate: http://goo.gl/3Cr1pw
+	 */
+	public TreeNode() {
+	}
+
+	/**
+	 *
+	 */
+	public TreeNode(String text) {
+		this.text = text;
+	}
+
+	/**
+	 * @return the text
+	 */
+	public String getText() {
+		return text;
+	}
+
+	/**
+	 * @param text
+	 *            the text to set
+	 */
+	public void setText(String text) {
+		this.text = text;
+	}
+
+	/**
+	 * @return the parentFolder
+	 */
+	public TreeFolder getParentFolder() {
+		return parentFolder;
+	}
+
+	/**
+	 * @param parentFolder the parentFolder to set
+	 */
+	public void setParentFolder(TreeFolder parentFolder) {
+		this.parentFolder = parentFolder;
+	}
+
+	/**
+	 * @return the index
+	 */
+	public int getIndex() {
+		return index;
+	}
+
+	/**
+	 * @param index the index to set
+	 */
+	public void setIndex(int index) {
+		this.index = index;
+	}
+
+	/**
+	 * @return the root
+	 */
+	public boolean isRoot() {
+		return root;
+	}
+
+	/**
+	 * @param root the root to set
+	 */
+	public void setRoot(boolean root) {
+		this.root = root;
+	}
+
+	/**
+	 * @return the leaf
+	 */
+	public boolean isLeaf() {
+		return leaf;
+	}
+
+	/**
+	 * @param leaf the leaf to set
+	 */
+	public void setLeaf(boolean leaf) {
+		this.leaf = leaf;
+	}
+
+	/**
+	 * @return the checked
+	 */
+	public Boolean isChecked() {
+		return checked;
+	}
+
+	/**
+	 * @param checked the checked to set
+	 */
+	public void setChecked(Boolean checked) {
+		this.checked = checked;
+	}
+
+	/**
+	 * @return the expandable
+	 */
+	public boolean isExpandable() {
+		return expandable;
+	}
+
+	/**
+	 * @param expandable the expandable to set
+	 */
+	public void setExpandable(boolean expandable) {
+		this.expandable = expandable;
+	}
+
+	/**
+	 * @return the expanded
+	 */
+	public boolean isExpanded() {
+		return expanded;
+	}
+
+	/**
+	 * @param expanded the expanded to set
+	 */
+	public void setExpanded(boolean expanded) {
+		this.expanded = expanded;
+	}
+
+	/**
+	 * @return the icon
+	 */
+	public String getIcon() {
+		return icon;
+	}
+
+	/**
+	 * @param icon the icon to set
+	 */
+	public void setIcon(String icon) {
+		this.icon = icon;
+	}
+
+	/**
+	 * @return the iconCls
+	 */
+	public String getIconCls() {
+		return iconCls;
+	}
+
+	/**
+	 * @param iconCls the iconCls to set
+	 */
+	public void setIconCls(String iconCls) {
+		this.iconCls = iconCls;
+	}
+
+	/**
+	 * @return the qTip
+	 */
+	public String getqTip() {
```

Comment:
Is this named correctly? Even though it looks weird, I'd expect it to be called with a capital `Q`, e.g. `getQTip`?